File: Python/code/T28/test_T28.py
import os
import Test1.test as file1
import Test2.test as file2
from Test1.test import N

files_to_compare = [
    ("Test1/test.py", "Test1_Nuitka/test.py"),
    ("Test2/test.py", "Test2_Nuitka/test.py")
]

def test_files_identical():
    for orig, nuitka in files_to_compare:
        assert os.path.exists(orig), f"Original file not found: {orig}"
        assert os.path.exists(nuitka), f"Nuitka file not found: {nuitka}"

        with open(orig, "r", encoding="utf-8") as f1:
            content1 = f1.read()
        with open(nuitka, "r", encoding="utf-8") as f2:
            content2 = f2.read()

        assert content1 == content2, f"Files differ: {orig} != {nuitka}"

def test_identical_output_check_all():
    array = [i % 2 for i in range(N)]
    ierror = 0
    copy1 = list(array)
    copy2 = list(array)
    result1 = file1.test(copy1, ierror)
    result2 = file2.test(copy2, ierror)

    for name, r in [("file1", result1), ("file2", result2)]:
        assert r is not None, f"{name}.test returned None!"
        assert isinstance(r, int), f"{name}.test should return a number, got {type(r)}"

    if not (copy1 == copy2):
        raise AssertionError(f"Mismatch detected between versions!")

    if not (result1 == result2):
        raise AssertionError(f"Mismatch detected between versions!")

if __name__ == "__main__":
    test_files_identical()
    test_identical_output_check_all()

File: Java/code/T28/test_T28.java
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Random;

public class test_T28 {

    public static void main(String[] args) {
        test_files_identical();
        test_identical_output_check_all();
    }

    public static void test_files_identical() {
        String[][] filesToCompare = {
            {"Test1/test.java", "Test1_Xint/test.java"},
            {"Test2/test.java", "Test2_Xint/test.java"}
        };

        for (String[] pair : filesToCompare) {
            try {
                Path pathOrig = Paths.get(pair[0]);
                Path pathXint = Paths.get(pair[1]);

                if (!Files.exists(pathOrig) || !Files.exists(pathXint)) {
                    System.err.println("[ERROR] File not found: " + pair[0] + " or " + pair[1]);
                    System.exit(1);
                }

                java.util.List<String> f1Lines = Files.readAllLines(pathOrig);
                java.util.List<String> f2Lines = Files.readAllLines(pathXint);

                f1Lines.removeIf(line -> ignoreLine(line));
                f2Lines.removeIf(line -> ignoreLine(line));

                if (!f1Lines.equals(f2Lines)) {
                    System.err.println("[ERROR] Files differ (ignoring package & REPS): "
                            + pair[0] + " != " + pair[1]);
                    System.exit(1);
                }

            } catch (IOException e) {
                System.err.println("[ERROR] Error reading files: " + e.getMessage());
                System.exit(1);
            }
        }
    }

    private static boolean ignoreLine(String line) {
        String l = line.trim();

        return l.startsWith("package ")
            || l.startsWith("public static final int REPS")
            || l.startsWith("public static final long REPS")
            || l.contains("for (int reps")
            || l.contains("for (long reps");
    }

    public static void test_identical_output_check_all() {
        try {
            int n = Test1.test.N;

            int[] array = new int[n];
            for (int i = 0; i < n; i++) {
                array[i] = i % 2;
            }

            int ierror = 0;

            int[] copy1 = Arrays.copyOf(array, array.length);
            int[] copy2 = Arrays.copyOf(array, array.length);

            int result1 = Test1.test.test(copy1, ierror);
            int result2 = Test2.test.test(copy2, ierror);

            if (!Arrays.equals(copy1, copy2)) {
                throw new AssertionError("Mismatch detected between versions (arrays differ)!");
            }

            if (result1 != result2) {
                throw new AssertionError("Mismatch detected between versions (results differ)!");
            }

        } catch (Throwable e) {
            System.err.println("[ERROR] Execution error: " + e.getMessage());
            e.printStackTrace();
            System.exit(1);
        }
    }
}